skunkworks_crow/src/main/java/org/odk/share/views/ui/bluetooth/BtReceiverActivity.java

```diff
@@ -0,0 +1,193 @@
+package org.odk.share.views.ui.bluetooth;
+
+import android.app.ProgressDialog;
+import android.bluetooth.BluetoothAdapter;
+import android.bluetooth.BluetoothDevice;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.Toast;
+
+import androidx.appcompat.widget.Toolbar;
+import androidx.recyclerview.widget.DividerItemDecoration;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import org.odk.share.R;
+import org.odk.share.bluetooth.BluetoothReceiver;
+import org.odk.share.bluetooth.BluetoothUtils;
+import org.odk.share.events.BluetoothEvent;
+import org.odk.share.events.DownloadEvent;
+import org.odk.share.rx.RxEventBus;
+import org.odk.share.rx.schedulers.BaseSchedulerProvider;
+import org.odk.share.services.ReceiverService;
+import org.odk.share.views.ui.common.injectable.InjectableActivity;
+
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.disposables.Disposable;
+import timber.log.Timber;
+
+/**
+ * Receive activity, for testing, needs refactor.
+ *
+ * @author huangyz0918 (huangyz0918@gmail.com)
+ */
+public class BtReceiverActivity extends InjectableActivity implements
+        BluetoothReceiver.BluetoothReceiverListener, BluetoothListAdapter.OnDeviceClickListener {
+
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+
+    @BindView(R.id.btn_refresh)
+    Button btnRefresh;
+
+    @BindView(R.id.list_bt_device)
+    RecyclerView recyclerView;
+
+    @Inject
+    ReceiverService receiverService;
+
+    @Inject
+    RxEventBus rxEventBus;
+
+    @Inject
+    BaseSchedulerProvider schedulerProvider;
+
+    private BluetoothReceiver bluetoothReceiver;
+    private final BluetoothListAdapter bluetoothListAdapter = new BluetoothListAdapter(this);
+    private boolean isConnected = false;
+    private ProgressDialog progressDialog;
+
+    private final CompositeDisposable compositeDisposable = new CompositeDisposable();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_bt_receive);
+        ButterKnife.bind(this);
+        initEvents();
+
+        setTitle(getString(R.string.connect_bluetooth_title));
+        setSupportActionBar(toolbar);
+
+        // checking for if bluetooth enabled
+        if (!BluetoothUtils.isBluetoothEnabled()) {
+            BluetoothUtils.enableBluetooth();
+            bluetoothListAdapter.updateDeviceList();
```

Comment:
The `updateDevieList()` should be present here in this activity.

Reply:
fixed.